js/src/common/utils/subclassOf.js

```diff
@@ -0,0 +1,6 @@
+/**
+ * Check if class A is a subclass of class B.
+ */
+export default function subclassOf(A, B) {
+  return A == B || A.prototype instanceof B;
```

Comment:
I think we want to check that `A` isn't falsy here before continuing, to reduce code duplication & random errors that originate here.

```suggestion
  return A && (A == B || A.prototype instanceof B);
  ```
```

Reply:
Good point!